src/scicookie/{{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/cli.py

```diff
@@ -0,0 +1,94 @@
+"""Module with CLI functions."""
+
+{%- if cookiecutter.command_line_interface == "Argparse" %}
+
+import argparse
+
+from {{ cookiecutter.package_slug }} import __version__
+{%- if cookiecutter.use_black == "yes" %}
+
+{% endif %}
+class CustomHelpFormatter(argparse.RawTextHelpFormatter):
+    """Formatter for generating usage messages and argument help strings.
+
+    Only the name of this class is considered a public API. All the methods
+    provided by the class are considered an implementation detail.
+    """
+
+    def __init__(
+        self,
+        prog,
+        indent_increment=2,
+        max_help_position=4,
+        width=None,
+        **kwargs,
+    ):
+        super().__init__(
+            prog,
+            indent_increment=indent_increment,
+            max_help_position=max_help_position,
+            width=width,
+            **kwargs,
+        )
+
+
+def get_args():
+    """Return the arguments for the CLI."""
+    parser = argparse.ArgumentParser(
+        prog="{{ cookiecutter.project_slug }}",
+        description=("{{ cookiecutter.project_name }}"),
+        epilog=(
+            "If you have any problem, open an issue at: "
+            "{{ cookiecutter.project_url }}"
+        ),
+        add_help=True,
+        formatter_class=CustomHelpFormatter,
+    )
+    parser.add_argument(
+        "--version",
+        action="store_true",
+        help="Show the version of the installed {{ cookiecutter.project_name }} tool.",
+    )
+
+    return parser
+
+
+def show_version():
+    """Show the version for the application."""
+    print(__version__)
+
+
+def app():
+    """Run the application."""
+    args_parser = get_args()
+    args = args_parser.parse_args()
+
+    if args.version:
+        return show_version()
+
+{%- elif cookiecutter.command_line_interface == "Click" %}
+
+import click
+
+from {{ cookiecutter.package_slug }} import __version__
+{%- if cookiecutter.use_black == "yes" %}
+
+{% endif %}
+@click.command()
+@click.option(
+    "--version",
+    is_flag=True,
+    help="Show the version of the installed {{ cookiecutter.project_name }} tool.",
+)
+def app(version):
+    """Run the application."""
+    if version:
+        return click.echo(__version__)
+    click.echo("You can add more Click commands here.")
+{%- if cookiecutter.use_black == "yes" %}
+
+{% endif %}
+if __name__ == "__main__":
```

Comment:
I think that these two lines should be at the end. It should be used by any of the CLI options.

Reply:
I think that this endif should be placed before the `if`.

It should be used for any selected CLI

Reply:
@YurelyCamacho , a friendly reminder about this issue

Reply:
@YurelyCamacho as you can see in line 88 ... that `if` condition is inside for click.
so if the user selected the argparse option, it will not be available for that.


does it make sense?

so the endif in line 91 should be moved up before the `if` condition in line 88

Reply:
@xmnlab This condition is for black, and I removed it in the last commit.

Reply:
Sorry .. maybe I am not explain myself very well.


cli.py will have a public function called app, and you created an app function for both CLI option. But you have just one `if __name__ == "__main__": app()` .. that should be implemented for both cases. So instead of duplicate the same python condition .. move that to outside of the jinja2 condition.

Let me know if that makes more sense now